debugging/coreclr/dockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,250 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { CancellationToken, DebugConfiguration, DebugConfigurationProvider, ProviderResult, WorkspaceFolder } from 'vscode';
+import { PlatformOS } from '../../utils/platform';
+import { DebugSessionManager } from './debugSessionManager';
+import { DockerManager, LaunchResult } from './dockerManager';
+import { FileSystemProvider } from './fsProvider';
+import { NetCoreProjectProvider } from './netCoreProjectProvider';
+import { OSProvider } from './osProvider';
+import { Prerequisite } from './prereqManager';
+
+interface DockerDebugBuildOptions {
+    context?: string;
+    dockerfile?: string;
+    tag?: string;
+    target?: string;
+}
+
+interface DockerDebugRunOptions {
+    containerName?: string;
+    os?: PlatformOS;
+}
+
+interface DebugConfigurationBrowserBaseOptions {
+    enabled?: boolean;
+    command?: string;
+    args?: string;
+}
+
+interface DebugConfigurationBrowserOptions extends DebugConfigurationBrowserBaseOptions {
+    windows?: DebugConfigurationBrowserBaseOptions;
+    osx?: DebugConfigurationBrowserBaseOptions;
+    linux?: DebugConfigurationBrowserBaseOptions;
+}
+
+interface DockerDebugConfiguration extends DebugConfiguration {
+    appFolder?: string;
+    appOutput?: string;
+    appProject?: string;
+    dockerBuild?: DockerDebugBuildOptions;
+    dockerRun?: DockerDebugRunOptions;
+}
+
+export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
+    constructor(
+        private readonly debugSessionManager: DebugSessionManager,
+        private readonly dockerManager: DockerManager,
+        private readonly fsProvider: FileSystemProvider,
+        private readonly osProvider: OSProvider,
+        private readonly netCoreProjectProvider: NetCoreProjectProvider,
+        private readonly prerequisite: Prerequisite) {
+    }
+
+    public provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken): ProviderResult<DebugConfiguration[]> {
```

Comment:
You should wrap stuff that gets called directly from vscode with callWithTelemetryAndErrorHandling.  E.g.:

        await callWithTelemetryAndErrorHandling('getCustomRegistryUsernamePwd', async function (this: IActionContext): Promise<void> {
              this.rethrowError = true;
              …

Reply:
Not needed for commands because our registerCommand handles it automatically.

Reply:
Wrapped call with `callWithTelemetryAndErrorHandling()`.